### DISCUSSION TITLE
Proxy not working for Clerk in Next.js 14 app on Vercel (and locally)?

### DISCUSSION THREAD

**Discussion by lancejpollard** [clerk/javascript#3076] [Support request - DO NOT OPEN HERE]
Cross-posting this StackOverflow question for visibility: _[How to add proxy for Clerk auth in Next.js/Vercel app?](https://stackoverflow.com/questions/78253752/how-to-add-proxy-for-clerk-auth-in-next-js-vercel-app)_

I am trying to have a multiple domain system following along with https://github.com/clerk/clerk-nextjs-multi-domain-example but not able to figure out yet how to get the proxy working (locally or in production on Vercel) for Next.js 14 using the app router system.

Would also help to know what [this `middleware.ts`](https://github.com/clerk/clerk-nextjs-multi-domain-example/blob/eb51a37ec0dc77f9a66bbd46e95202f3e13723f4/apps/primary/middleware.ts) is doing (why the complicated matcher?):

```ts
import { authMiddleware } from "@clerk/nextjs/server";

export default authMiddleware();

export const config = {
  matcher: ["/((?!.*\\..*|_next).*)", "/", "/(api)(.*)"],
};
```

If the proxy system can't work on Next.js 14 / Vercel (https://clerk.com/docs/advanced-usage/using-proxies), how can I get it working with Next.js / Vercel in some other fashion, such as integrating CloudFlare or something else? Hopefully it's not too involved 👍

**Comment by jescalan** [comment]
please open this issue with [clerk support](https://clerk.com/contact/support) instead!
